<?xml version="1.0"?>

<launch>

  <arg name="geotiff_map_file_path" default="$(find hector_geotiff)/maps"/>

  <param name="/use_sim_time" value="false"/>

  <node name="static_tf0" pkg="tf" type="static_transform_publisher" args="1 0 0 0 0 0  /world /map 100"/>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find hector_slam_launch)/rviz_cfg/mapping_demo.rviz"/>

  <include file="$(find ros-x-habitat)/launch/mapping_default.launch"/>

<!--   <include file="$(find hector_geotiff)/launch/geotiff_mapper.launch"> -->
<!--     <arg name="trajectory_source_frame_name" value="base_frame"/> -->
<!--     <arg name="map_file_path" value="$(arg geotiff_map_file_path)"/> -->
<!--   </include> -->


  <!-- Use RGBD synchronization -->
  <!-- Here is a general example using a standalone nodelet,
       but it is recommended to attach this nodelet to nodelet
       manager of the camera to avoid topic serialization -->
  <node pkg="nodelet" type="nodelet" name="rgbd_sync" args="standalone rtabmap_ros/rgbd_sync" output="screen">
    <remap from="rgb/image"       to="ros_img_rgb"/>
    <remap from="depth/image"     to="ros_img_depth"/>
    <remap from="rgb/camera_info" to="camera_info_topic"/>
    <remap from="rgbd_image"      to="rgbd_image"/> <!-- output -->
    <!-- Should be true for not synchronized camera topics
         (e.g., false for kinectv2, zed, realsense, true for xtion, kinect360)-->
    <param name="approx_sync"       value="true"/>
  </node>
  <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="--delete_db_on_start">
        <param name="frame_id" type="string" value="base_frame"/>
        <param name="subscribe_depth" type="bool" value="false"/>
        <param name="subscribe_rgbd" type="bool" value="true"/>
        <param name="subscribe_scan" type="bool" value="true"/>
        <remap from="odom" to="odom"/>
        <remap from="scan" to="scan"/>
        <remap from="rgbd_image" to="rgbd_image"/>
        <param name="queue_size" type="int" value="10"/>
        <!-- RTAB-Map's parameters -->
        <param name="RGBD/NeighborLinkRefining" type="string" value="true"/>
        <param name="RGBD/ProximityBySpace"     type="string" value="true"/>
        <param name="RGBD/AngularUpdate"        type="string" value="0.01"/>
        <param name="RGBD/LinearUpdate"         type="string" value="0.01"/>
        <param name="RGBD/OptimizeFromGraphEnd" type="string" value="false"/>
        <param name="Grid/FromDepth"            type="string" value="false"/> <!-- occupancy grid from lidar -->
        <param name="Reg/Force3DoF"             type="string" value="true"/>
        <param name="Reg/Strategy"              type="string" value="1"/> <!-- 1=ICP -->
        <!-- ICP parameters -->
        <param name="Icp/VoxelSize"                 type="string" value="0.05"/>
        <param name="Icp/MaxCorrespondenceDistance" type="string" value="0.1"/>
  </node>

</launch>

